### DISCUSSION TITLE
Range expression in case?

### DISCUSSION THREAD

**Discussion by vanillajonathan** [PRQL/prql#3653] [Ideas]
What if you could use a range expression such as `90..110` in a `case` expression?

**Comment by eitsupi** [comment]
Maybe you need this? #3461

**Reply by vanillajonathan** [reply]
Maybe

**Comment by eitsupi** [comment]
Closing in favor of #3461

**Reply by vanillajonathan** [reply]
Would this allow something like this?

```elm
from athletes
derive stature = case {
  height == 180..200 => "tall",
  height == 201..250 => "super-tall",
}
```

**Reply by eitsupi** [reply]
I may not have understood your intent.
But isn't this already working?

```elm
from athletes
derive stature = case {
  (height | in 180..200) => "tall",
  (height | in 201..250) => "super-tall",
}
```

```sql
SELECT
  *,
  CASE
    WHEN height BETWEEN 180 AND 200 THEN 'tall'
    WHEN height BETWEEN 201 AND 250 THEN 'super-tall'
    ELSE NULL
  END AS stature
FROM
  athletes

-- Generated by PRQL compiler version:0.9.5 (https://prql-lang.org)
```

**Reply by vanillajonathan** [reply]
Ah, yes that works great! I just wasn't aware of the syntax.

The syntax isn't as obvious to me as the one in Rust. Here I have to use parenthesis, pipe and `in`, but perhaps that is just me not being used to it yet.
https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#matching-ranges-of-values-with-

I kind of would like to do:
```elm
from athletes
derive stature = match athletes.height  {
  180..200 => "tall",
  201..250 => "super-tall",
}
```
This way I don't have to repeat `height` for every branch, and I wouldn't be surprised where `height` comes from, as it would be a property of of `athletes`, hence I could also get column auto-completion for it.

**Reply by eitsupi** [reply]
There is a long discussion here #1286, and I don't think the equivalent of a "simple" `CASE WHEN` in SQL that you want has been implemented yet.
